File: back-end/domains/latihan/request.go
package latihan

type LatihanReq struct {
	Name        string   `json:"name" form:"name" binding:"required"`
	SoalLatihan []string `json:"soal_latihan" form:"soal_latihan" binding:"required"`
}

type DeleteLatihanReq struct {
	ID string `json:"id" binding:"required"`
}

type GetAllLatihanReq struct {
	Page  int64
	Limit int64
}

type StatsLatihanReq struct {
	LatihanID string  `json:"latihan_id" binding:"required"`
	Score     float64 `json:"score" binding:"required"`
}

File: mobile/app/src/main/java/com/example/signlink/data/models/latihan/Request.kt
package com.example.signlink.data.models.latihan

import com.google.gson.annotations.SerializedName

data class LatihanReq(
    val name: String,
    @SerializedName("soal_latihan")
    val soalLatihan: List<String>
)

data class StatsLatihanReq(
    @SerializedName("latihan_id")
    val latihanId: String,
    val score: Float
)